File: Java/drop-every-other.java
/*  Cyrus Chen
    chen3736
 
    compile ->     javac hwk_01.java
    run ->         java hwk_01
    
            keep enter numbers, negatives will trigger output
*/

import java.util.*;



public class hwk_01 {

/*  main that only make calls to methods.*/
    public static void main(String[] args) {

        hwk_01.printResult(hwk_01.inputReader());
    }

/*  This method take user inputs and store in an array list of integers
    using Scanner class to take inputs   */
    public static ArrayList<Integer> inputReader(){

        int getint = 0;
        int loopend = 0;
        Scanner read = new Scanner(System.in);
        ArrayList<Integer> userInputs = new ArrayList<Integer>();

        while (loopend >= 0){
            System.out.println("Enter a number, end with negatives");
            getint = read.nextInt();
            if (getint >= 0){
                userInputs.add(getint);
            }
            else{
                loopend = -1;
            }
        }
        return userInputs;
    }
    
/*  This method loop through the array and store every other elements into
    a new array, then print out the new array.  */    
    
    public static void printResult(ArrayList<Integer> inputlist){
        
        ArrayList<Integer> outputlist = new ArrayList<Integer>();
        
        for(int i=0; i<inputlist.size(); i++){
            if (i % 2 == 0){
                outputlist.add(inputlist.get(i));
            }
        }
        System.out.println(outputlist);
    }
}

File: C/drop-every-other.c

/*******************************************************************************
 * Name            : hwk_01.c
 * Project         : CSCI 5106
 * Description     : Implementation of input reader utility
 * Author          : Cyrus Chen
 ******************************************************************************/

//    compile ->     gcc hwk_01.c
//    run ->         ./a.out
//        keep enter numbers, negatives will trigger output



#include <stdio.h>
#include <stdlib.h>
// Uses a loop to take input, then print every other element from the input list


// inputReader ask the user to input integer, and store the input til negative
// take a pointer to int, modify the value and return the number of elements
int inputReader(int* inputlist) {
    int userinput = 0;
    int counter = 0;
    while(1){       // loop
        printf("Enter a number, end with negatives ");
        if (scanf("%d",&userinput) == 0) {
        printf("Invalid input, integers please \n");
            exit(0);
            
        } else {

            if (userinput >= 0) {   // when inputs >= 0
                inputlist[counter] = userinput; // store value
                counter++;
            } else break;   // negative input, break out from the loop
        }
    }
    
    
    return counter;
}

// printResult takes a pointer of int and the number of elements.
// print every other element in that pointer
int printResult(int* inputlist, int counter) {
    int i = 0;
    for(; i < counter; i++) {
        if (i % 2 == 0) {   // if even, since i starts with 0
        printf("%d ",inputlist[i]);
        }
    }
    return 0;
}



int main(int argc, char** argv) {
    int *userinput = malloc(sizeof(int));   //malloc memory for input
    int counter = inputReader(userinput);
    printResult(userinput, counter);
    free(userinput);
    printf("\n");
    return 0;
}
